### ISSUE TITLE
datalad archives special remotes should have a non-shared config to immediately drop data

### ISSUE THREAD

**Issue by adswa** [datalad/datalad#7121]
This takes a note of something thought out loud by @bpoldrack during the office hour: The archive special remote should take a local config in .git/config to drop the archive right away. @bpoldrack can probably elaborate a bit more.

**Comment by bpoldrack** [comment]
Yes. The point is, that this about dropping/cleaning the temporary space after getting a file. Currently this requires a `get` followed by a `clean`. If you need to get a lot of files across a hierarchy of datasets and parallelize things, `clean` after every file can be a troublesome approach. However, a switch for `get` can't be a solution, since the special remote doesn't know what was the annex command, let alone the datalad command, that it was called from. A "regular" special remote config is no solution either, since it would be committed in the `git-annex` branch and affect other users.
However, git-annex special remote supports local configs for special remotes for a while now. This could be the mechanism to allow such a feature.

**Comment by yarikoptic** [comment]
FWIW I think it is a good idea!  May be could be generalized into something like `archives-auto-drop-after` which could be "immediately" (which would be as it was just a bool flag) or could be "day" or some other duration so whenever some codepath gets there - it would trigger analysis and GC of those archives. It is more elaborate but could be nice... or could start to be implemented just with "immediate" but not making it a bool flag thus keeping the door open for other strategies.

**Comment by mih** [comment]
Clarified that "non-shared" means "local-scope" configuration.
